astropy/visualization/wcsaxes/coordinate_helpers.py

```diff
@@ -145,6 +147,191 @@ def __init__(
+            AstropyDeprecationWarning,
+        )
+        self._coord_type = value
+
+    @property
+    def coord_unit(self):
+        """
+        The unit that this coordinate is in given the output of transform.
+        """
+        return self._coord_unit
+
+    @coord_unit.setter
+    def coord_unit(self, value):
+        warnings.warn(
+            "Setting CoordinateHelper.coord_unit directly is deprecated",
+            AstropyDeprecationWarning,
+        )
+        self._coord_unit = value
+
+    @property
+    def coord_wrap(self):
+        """
+        The angle at which the longitude wraps (defaults to 360 degrees).
+        """
+        return self._coord_wrap
+
+    @coord_wrap.setter
+    def coord_wrap(self, value):
+        warnings.warn(
+            "Setting CoordinateHelper.coord_wrap directly is deprecated, use CoordinateHelper.set_coord_type instead",
+            AstropyDeprecationWarning,
+        )
+        self._coord_wrap = value
+
+    @property
+    def frame(self):
+        """
+        The frame of the :class:`~astropy.visualization.wcsaxes.WCSAxes`.
+        """
+        return self._frame
+
+    @frame.setter
+    def frame(self, value):
+        warnings.warn(
+            "Setting CoordinateHelper.frame directly is deprecated",
+            AstropyDeprecationWarning,
+        )
+        self._frame = value
+
+    @property
+    def default_label(self):
+        """
+        The axis label to show by default if none is set later.
+        """
+        return self._default_label
+
+    @default_label.setter
+    def default_label(self, value):
+        warnings.warn(
+            "Setting CoordinateHelper.default_label directly is deprecated",
+            AstropyDeprecationWarning,
+        )
+        self._default_label = value
+
+    @property
+    def ticks(self):
+        warnings.warn(
+            "CoordinateHelper.ticks should not be accessed directly and is deprecated",
+            AstropyDeprecationWarning,
+        )
+        return self._ticks
+
+    @ticks.setter
+    def ticks(self, value):
+        warnings.warn(
+            "Setting CoordinateHelper.ticks directly is deprecated",
+            AstropyDeprecationWarning,
+        )
+        self._ticks = value
+
+    @property
+    def ticklabels(self):
+        warnings.warn(
+            "CoordinateHelper.ticklabels should not be accessed directly and is deprecated",
+            AstropyDeprecationWarning,
+        )
+        return self._ticks
+
+    @ticklabels.setter
+    def ticklabels(self, value):
+        warnings.warn(
+            "Setting CoordinateHelper.ticklabels directly is deprecated",
+            AstropyDeprecationWarning,
+        )
+        self._ticklabels = value
+
+    @property
+    def axislabels(self):
+        warnings.warn(
+            "CoordinateHelper.ticks should not be accessed directly and is deprecated",
```

Comment:
This is a typo,. it should be `CoordinateHelper.axislabels`.

Reply:
Also, I didn't notice this until I was testing something else out with astropy main, but dkist is using this: https://github.com/DKISTDC/dkist/blob/main/dkist/dataset/tiled_dataset.py#L165 not sure if there's a better way to do it without relying on this?

Reply:
#17199 👀

Reply:
And https://github.com/astropy/astropy/pull/16686 for the replacement for getting visible axes